pkg/server/interceptors/errors.go

```diff
@@ -0,0 +1,46 @@
+package interceptors
+
+import (
+	"context"
+	"errors"
+
+	grpczap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	ErrInternalServer       = errors.New("internal server error")
+	grpcInternalServerError = status.Errorf(codes.Internal, ErrInternalServer.Error())
+)
+
+// UnaryErrorHandler is a unary server interceptor that captures the request and response and overrides the error message
+func UnaryErrorHandler() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp any, err error) {
+		resp, err = handler(ctx, req)
+		if err != nil {
+			return nil, errWrapper(ctx, err)
+		}
+		return resp, nil
+	}
+}
+
+func errWrapper(ctx context.Context, err error) error {
+	grpczap.Extract(ctx).Error(err.Error())
+
+	// check if grpc status error
+	if _, ok := status.FromError(err); ok {
```

Comment:
Doesn't `status.FromError` handle `context.canceled` as a known error and return `ok`?

Reply:
It doesn't handle it, there is a test to verify it as well in `pkg/server/interceptors/errors_test.go`